### PR TITLE
Added the exists method to the FileSystem.

### PR DESCRIPTION
**Pull request by sunildixit**
This PR adds the exists(string filePath) api to the FileSystem, which will allow the users to check whether a given folder or file exists in the file system or not.

I'll be adding more methods once I know, that I have done this work correctly.

**Example: Check for an existing file**

```cs
SparkSession spark = ....
FileSystem fs = FileSystem.Get(spark.SparkContext.HadoopConfiguration());
bool fileExists = fs.Exists("abfss://container@datalake.dfs.core.windows.net/myfolder/file.csv");
```

This PR relates to #328

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/spark?pullRequest=909) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/spark?pullRequest=909) before we can accept your contribution.<br/><br/>:x: sunildixit [sign now](https://cla.dotnetfoundation.org/dotnet/spark?pullRequest=909)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/spark?pullRequest=909) it.</sub>

**Comment by sunildixit** [comment]
@AFFogarty Sorry it took me some time to get this PR out. Please take a look and let me know.

File: src/csharp/Microsoft.Spark.E2ETest/Hadoop/FileSystemTests.cs [hunk 1]
```diff
@@ -52,5 +53,28 @@ public void TestDelete()
...
+
+        [Fact]
+        public void TestExists()
+        {
+            using FileSystem fs = FileSystem.Get(_spark.SparkContext.HadoopConfiguration());
+
+            using var tempDirectory = new TemporaryDirectory();
+            string path = Path.Combine(tempDirectory.Path, "temp-table");
+
+            _spark
```
Comment (AFFogarty) [inline src/csharp/Microsoft.Spark.E2ETest/Hadoop/FileSystemTests.cs:65]: Does this fit on one line?

File: src/csharp/Microsoft.Spark.E2ETest/Hadoop/FileSystemTests.cs [hunk 2]
```diff
@@ -52,5 +53,28 @@ public void TestDelete()
...
+                .Write()
+                .Csv(path);
+
+            Assert.True(fs.Exists(path));
+
+            var dataFile = Directory.GetFiles(path, "*.csv").FirstOrDefault();
+
+            Assert.NotNull(dataFile);
+
+            Assert.True(fs.Exists(dataFile));
```
Comment (AFFogarty) [inline src/csharp/Microsoft.Spark.E2ETest/Hadoop/FileSystemTests.cs:77]: Maybe we should add an assertion to test that `fs.Exists()` returns `false` on a non-existent path?

For example you could do something like this:

```
string path = Path.Combine(tempDirectory.Path, "temp-table");

Assert.False(fs.Exists(path));
_spark.Range(25).Write().Csv(path);
Assert.True(fs.Exists(path));
```

That way, you prove that the `Exists()` return value changes after writing to that path.

File: src/csharp/Microsoft.Spark/Hadoop/Fs/FileSystem.cs [hunk 3]
```diff
@@ -55,6 +55,14 @@ public bool Delete(string path, bool recursive = true)
             return (bool)_jvmObject.Invoke("delete", pathObject, recursive);
         }
 
+        public bool Exists(string path)
```
Comment (AFFogarty) [inline src/csharp/Microsoft.Spark/Hadoop/Fs/FileSystem.cs:58]: Could you please add a documentation comment describing the method?  You can copy its content from open-source `org.apache.hadoop.fs.FileSystem`.

File: src/csharp/Microsoft.Spark.E2ETest/Hadoop/FileSystemTests.cs [hunk 4]
```diff
@@ -52,5 +53,28 @@ public void TestDelete()
 
             Assert.False(Directory.Exists(path));
         }
+
+        [Fact]
+        public void TestExists()
```
Comment (AFFogarty) [inline src/csharp/Microsoft.Spark.E2ETest/Hadoop/FileSystemTests.cs:58]: Could you please add a documentation comment describing the test?

**Review by AFFogarty — COMMENTED** [review]
Thank you for your PR, @sunildixit.  I have left a few comments. 😃

File: src/csharp/Microsoft.Spark.E2ETest/Hadoop/FileSystemTests.cs (hunk 4 above)
Comment (sunildixit) [reply src/csharp/Microsoft.Spark.E2ETest/Hadoop/FileSystemTests.cs:62]: Added the documentation about the test

File: src/csharp/Microsoft.Spark.E2ETest/Hadoop/FileSystemTests.cs (hunk 2 above)
Comment (sunildixit) [reply src/csharp/Microsoft.Spark.E2ETest/Hadoop/FileSystemTests.cs:79]: Great catch. I have added the assertion.

File: src/csharp/Microsoft.Spark.E2ETest/Hadoop/FileSystemTests.cs (hunk 1 above)
Comment (sunildixit) [reply src/csharp/Microsoft.Spark.E2ETest/Hadoop/FileSystemTests.cs]: I have added it to one line.

File: src/csharp/Microsoft.Spark.E2ETest/Hadoop/FileSystemTests.cs (hunk 2 above)
Comment (sunildixit) [reply src/csharp/Microsoft.Spark.E2ETest/Hadoop/FileSystemTests.cs:79]: Great catch. I have added the check.

File: src/csharp/Microsoft.Spark/Hadoop/Fs/FileSystem.cs (hunk 3 above)
Comment (sunildixit) [reply src/csharp/Microsoft.Spark/Hadoop/Fs/FileSystem.cs:63]: Done

**Review by AFFogarty — APPROVED** [review]
LGTM. Thanks @sunildixit !

**Review by Niharikadutta — DISMISSED** [review]
LGTM, thanks @sunildixit !

**Review by suhsteve — DISMISSED** [review]
LGTM.

File: src/csharp/Microsoft.Spark.E2ETest/Hadoop/FileSystemTests.cs [hunk 5]
```diff
@@ -52,5 +53,30 @@ public void TestDelete()
...
+
+            string path = Path.Combine(tempDirectory.Path, "temp-table");
+
+            Assert.False(fs.Exists(path));
+
+            _spark.Range(25).Coalesce(1).Write().Csv(path);
+
+            Assert.True(fs.Exists(path));
+
+            var dataFile = Directory.GetFiles(path, "*.csv").FirstOrDefault();
```
Comment (imback82) [inline src/csharp/Microsoft.Spark.E2ETest/Hadoop/FileSystemTests.cs:76]:
```suggestion
            string dataFile = Directory.GetFiles(path, "*.csv").FirstOrDefault();
```

**Review by imback82 — COMMENTED** [review]
LGTM, thanks @sunildixit!

**Pull request merged**